### PR TITLE
Add auto-syncing useAddonState for manager & preview

### PR DESCRIPTION
**Pull request by ndelangen**
@Joseph-Whiunui

### CODE REVIEW

**Comment by ndelangen** [comment]
@atanasster ping!

File: examples/dev-kits/config.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { configure } from '@storybook/react';
```
Comment (ndelangen) [inline examples/dev-kits/config.js]: bad

File: examples/dev-kits/webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+module.exports = async ({ config, mode }) => ({
```
Comment (ndelangen) [inline examples/dev-kits/webpack.config.js]: bad

**Comment by lifeiscontent** [comment]
@ndelangen any reasons for this being closed? I'd like to use Addon state in the preview!

**Pull request closed without merging**